scalableminds/webknossos
# Easier folder creation and configurable target folder for dataset upload

- add "Add Folder" button next to "Add Dataset" button
![image](https://user-images.githubusercontent.com/2486553/208424281-bf95ac6d-5f2d-4cd4-a29e-ba4114eadb06.png)
- dummy context menu when rightclicking into the blank space of the sidebar:
![image](https://user-images.githubusercontent.com/2486553/208424412-575ec092-19fd-468a-840f-30e10f152615.png)
- clicking "add dataset" will encode the active folder id in the url so that the import view can use that to initial the new target-folder selection
![image](https://user-images.githubusercontent.com/2486553/208424518-58b01588-06d2-466b-b4b8-61e5de80f955.png)
![image](https://user-images.githubusercontent.com/2486553/208424553-ade6bd99-e573-47e8-b811-76e6085c03b0.png)


~The target folder functionality is currently front-end only (it simply uses the API to move the DS after the upload). This won't work if the dataset needs to be converted first. Not sure whether this should block the PR. The backend part probably won't happen in this year. Merging this now could be confusing to users when the DS doesn't pop up in the desired folder. However, this can be confusing right now, too~

### URL of deployed dev instance (used for testing):
- https://folderfollowup.webknossos.xyz

### Steps to test:
- use context menu in sidebar on folders and on blank space
- use new add-folder button
  - use the target folder UI to change the folder (should have initialized to the folder that was active in the dashboard)
  - upload something (zarr or normal)
  - dataset should appear there

### Issues:
- contributes to #6669 

------
(Please delete unneeded items, merge only when none are left open)
